### ISSUE TITLE
did you tried this on ofxGuiExtended?

### ISSUE THREAD

**Issue by moebiussurfing** [npisanti/ofxGuiPresetSelector#1]
I tried to change the addon like this, but no succes:
//    vector<ofxPanel*>   guis;
    vector<ofxGuiPanel*>   guis;

i not sure if it's easy or not. ofxGuiExtended replaces ofxGui on the main repo version, not an addon to complement ofxGui anymore i think

**Comment by moebiussurfing** [comment]
...should be nice an addon to handle any king of presets (structs/vectors/ofGroupParameters) with matrix buttons, preset names and triggers.

**Comment by npisanti** [comment]
actually this addon just use the `saveToFile()` and `loadFromFile()` methods of ofxGui so as long as ofxGuiExtended has this methods it could be easy to modify for that, but yes, we really need some kind of ofxPresets addon

**Comment by moebiussurfing** [comment]
thanks @npisanti. just forked and switched from ofxGui to ofxGuiExtended.

I have one question: 
why did you based the preset saving into ofxGui besides ofParameterGroup? what's the point?
I am asking because may be tiny to use a group to include any gui addon or maybe to use without gui... I am missing something?

**Comment by npisanti** [comment]
ah no, you are totally right, probably when i made this addon i just didn't know about `ofSerialize` and `ofDeseriealize` (or maybe they didn't existed yet, i think ofJson is included in oF since 0.10). If i had to code this again i would totally go for ofParameterGroup serialized to JSON (but i don't need something like this addon at the moment, i don't even remember the last time i used it)

**Comment by moebiussurfing** [comment]
hey, i did the switch to ofParameterGroup already. 
https://github.com/moebiussurfing/ofxGuiPresetSelector
Thanks for your addon, has been very helpful, and i'll try now to customize more to add structs/vector and json.

I don't want to abuse your help, but if you don't mind I have one doubt... I had to avoid the use the ofParameterGroup as pointer. Maybe it's something about the nature of ofParameters... because they are shared / smart pointers or something (my skills limitations too). Nevermind about this. 

But what's the point of use (in general) the pointers inside a class (or when passing to)? 
It's a handy way to share variables from ofApp scope inside (and backwards) the class? 
(I have seen sometimes about singleton debates and etc, not sure if it's related.)

```
    // NOTES:

    // pointer
    // ofParameterGroup * params;
    // void add( ofParameterGroup & group, initializer_list<int> keysList );
    // void ofxGuiPresetSelector::add( ofParameterGroup & group, int numPresets ) {
    // vector<ofParameterGroup*> groups;
    // groups.push_back( &group );

    // no-pointer
    // ofParameterGroup params;
    // void add( ofParameterGroup group, initializer_list<int> keysList );
    // void ofxGuiPresetSelector::add( ofParameterGroup group, int numPresets ) {
    // vector<ofParameterGroup> groups;
    // groups.push_back( group );
```

**Comment by npisanti** [comment]
Yes it's just an handy way to have objects passed from the ofApp or from other classes. The use with singletons is a different tecnique.

I use a pointer so there is no copy, the pointer is pointing to the original object, otherwise a new objects will be allocated and the code will operate on that, and the changes will go to the internal variable and not to the external object that i was internally referening.

In your case as you are using ofSerialize and ofDeserialize it could work as long as you don't add parameters to the linked ofParameterGroup after adding it to the ofxGuiPresetSelector.
